stdlib/builtin.ncl

```diff
@@ -224,5 +224,39 @@
       ```
       "%
     = fun msg x => %trace% msg x,
+
+    FailWith
```

Comment:
I'm not a fan of `With` in the name here, as it doesn't tell us _what_ we're failing with. I think `FailWithMessage` is significantly clearer.

Reply:
The `With` probably comes from analogy with `contract.blame_with`. I'm not opposed to renaming `FailWith` to `FailWithMessage`, it just raises the question whether `contract.blame_with` (and `fail_with` right here) should also have their names revised.

Reply:
Ah good point! I guess you could infer this from what I've said above, but to be extra clear: I'm in favour of renaming both for clarity & consistency. :grinning:

Reply:
Let's do that bikeshedding later and merge it the way it is, just in the interest of getting this PR over with.